typewise_alert.test.py

```diff
@@ -1,11 +1,90 @@
+import random
 import unittest
 import typewise_alert
 
 
+# Have also performed Handle Comms tests here
 class TypewiseTest(unittest.TestCase):
-  def test_infers_breach_as_per_limits(self):
-    self.assertTrue(typewise_alert.infer_breach(20, 50, 100) == 'TOO_LOW')
+    def test_check_and_alert(self):
+        new_limits_for_cooling_types = {
+            "PASSIVE_COOLING": (-20, 20),
+            "MED_ACTIVE_COOLING": (0, 100),
+            "HI_ACTIVE_COOLING": (-1000, 2000),
+        }
 
+        def test_send_func(breach_type):
+            print(breach_type)
+            return f"TEST_FUNC,{breach_type}"
 
-if __name__ == '__main__':
-  unittest.main()
+        test_alerts = [
+            "TOO_LOW",
+            "TOO_HIGH",
+            "NORMAL",
+            "TOO_HIGH"
+        ]
+        test_alert_mechanisms = {
+            "TO_CONTROLLER": "CONTROLLER_MESSAGE",
+            "TO_EMAIL": "EMAIL",
+            "TO_CONSOLE": "CONSOLE_MESSAGE",
+            "TO_TEST_FUNC": "TEST_FUNC"
+        }
+        typewise_alert_objects = [
+            typewise_alert.TypewiseAlert(
+                limits_for_types=new_limits_for_cooling_types),
+            typewise_alert.TypewiseAlert(
+                alert_mail_details={"TOO_HIGH": {
+                    "recipient": "test_high_temp_expert@bosch.com",
+                    "email_message": "Temperature too high."
+                }}),
+            typewise_alert.TypewiseAlert(),
+            typewise_alert.TypewiseAlert(
+                alert_target_funcs={"TO_TEST_FUNC": test_send_func})
+        ]
+        test_limits_for_cooling_types = [
+            new_limits_for_cooling_types,
+            typewise_alert_objects[1].default_limits_for_cooling_types,
+            typewise_alert_objects[2].default_limits_for_cooling_types,
+            typewise_alert_objects[3].default_limits_for_cooling_types
+        ]
+        index = 0
+        for test_alert_mechanism in test_alert_mechanisms:
+            count_cooling_types = len(test_limits_for_cooling_types[index])
```

Comment:
This loop is hard to understand. Can you split it into functions with names, so that it reads like you would explain it?